synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -949,6 +965,14 @@ void Widget_Timetrack::on_waypoint_double_clicked(const Widget_Timetrack::Waypoi
 	}
 }
 
+void Widget_Timetrack::on_no_waypoint_clicked(unsigned int button, Gdk::Point button_cord)
+{
+	synfig::Time time= time_plot_data->get_t_from_pixel_coord(button_cord.get_x());
+	CanvasView::LooseHandle canvas_view = current_canvas_view; //alterntively we can include app.h to use get_selected_canvas_view() but I dont think its worth including just for this
```

Comment:
don't couple stuff like this :) You are creating a mutual dependency.
Create a new signal for `Widget_Timetrack` and emit it here.
There are already other signals to you follow the idea (widget_timetrack.h):
```c++
	sigc::signal<void, synfigapp::ValueDesc, std::set<synfig::Waypoint,std::less<synfig::UniqueID> >, int>& signal_waypoint_clicked() { return signal_waypoint_clicked_; }
	sigc::signal<void, synfigapp::ValueDesc, std::set<synfig::Waypoint,std::less<synfig::UniqueID> >, int>& signal_waypoint_double_clicked() { return signal_waypoint_double_clicked_; }
```
Don't forget to connect in `CanvasView`.

Reply:
Oh ok, I thought this way would be better than creating yet another signal, but yes as opposed to mutual dependency, new signal seems better. 

P.s. I just left home will do when I get back in a few hours as I don't have the laptop with me now